src/vs/code/electron-main/app.ts

```diff
@@ -620,6 +625,28 @@ export class CodeApplication extends Disposable {
 		return initialProtocolUrls;
 	}
 
+	private setupManagedRemoteResourceUrlHandler(mainProcessElectronServer: ElectronIPCServer) {
+		const notFound = (): Electron.ProtocolResponse => ({ statusCode: 404, data: 'Not found' });
+		const remoteResourceChannel = new Lazy(() => mainProcessElectronServer.getChannel(
+			NODE_REMOTE_RESOURCE_CHANNEL_NAME,
+			new NodeRemoteResourceRouter(),
+		));
+
+		protocol.registerBufferProtocol(Schemas.vscodeRemoteResource, (request, callback) => {
+			const url = URI.parse(request.url);
+			if (!url.authority.startsWith('window:')) {
+				return callback(notFound());
+			}
+
+			remoteResourceChannel.value.call<NodeRemoteResourceResponse>(NODE_REMOTE_RESOURCE_IPC_METHOD_NAME, [url]).then(
```

Comment:
I am not familiar with the details of the issue this PR is trying to solve, but just asking from Electron API adoption perspective. In the current implementation, when the renderer process encounters a resource with `vscode-managed-remote-resource` scheme, the main process is tasked with sending an ipc to renderer for resolving and loading the resource via some fileservice which then again streams the data back into the renderer via another ipc internally for the protocol handler implementation. 

Is there any way the resource can be read from the main process to make one less IPC call ?